test/unit/htmlValidatorTest.js

```diff
@@ -0,0 +1,507 @@
+/* eslint-env mocha */
+
+const assert = require('assert')
+const path = require('path')
+const generateTestApp = require('../util/generateTestApp')
+const cleanupTestApp = require('../util/cleanupTestApp')
+const fork = require('child_process').fork
+const fse = require('fs-extra')
+const request = require('supertest')
+
+describe('Roosevelt HTML Validator Test', function () {
+  this.timeout(20000)
+
+  // location of the test app
+  const appDir = path.join(__dirname, '../', 'app', '/htmlValidatorTest')
+
+  // options that would be put into generateTestApp params
+  const options = {rooseveltPath: '../../../roosevelt', method: 'startServer'}
+
+  beforeEach(function (done) {
+    // copy the mvc dir from util to the test app
+    fse.copySync(path.join(appDir, '../', '../', 'util', 'mvc'), path.join(appDir, 'mvc'))
+    done()
+  })
+
+  afterEach(function (done) {
+    cleanupTestApp(appDir, (err) => {
+      if (err) {
+        throw err
+      } else {
+        done()
+      }
+    })
+  })
+
+  it('should give back a validation error page if the htmlValidator is on and the app is trying to send back a html page with errors', function (done) {
+    // generate the app
+    generateTestApp({
+      generateFolderStructure: true,
+      appDir: appDir,
+      htmlValidator: {
+        enable: true
+      },
+      onServerStart: true
+    }, options)
+
+    // fork the app and run it as a child process
+    const testApp = fork(path.join(appDir, 'app.js'), ['--dev'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    // on the message that we get back that the server has started, test the htmlValidator by trying to recieve a bad html page
+    testApp.on('message', () => {
+      // request the bad html page
+      request('http://localhost:43711')
```

Comment:
don't hardcode the port on these. default could change some day which would require us to update all the tests